src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -455,6 +470,10 @@ private static void StartThread(IntPtr parameter)
             }
             finally
             {
+                // what if OS thread dies and this block never invokes?
```

Comment:
Is there also a problem with threads that wander in into the runtime (we take `InitializeExistingThread` path for them) and then somebody flips them to be a foreground thread? I do not see anything that would decrement the counter if they exit.

Reply:
I understood the problem you highlighted but I can't find any calls to `InitializeExistingThread` apart from `Thread.CurrentThread` and `ThreadPoolCallbackWrapper.Enter()` I guess second one is my buddy.

Reply:
`Thread.CurrentThread` path is the what will be hit for the threads that wandered in.